lib/records/fhir_dstu2.rb

```diff
@@ -332,6 +353,29 @@ def self.observation(observation, fhir_record, patient, encounter)
         fhir_record.entry << entry
       end
 
+      def self.quality_of_life_observation(observation, fhir_record, patient)
```

Comment:
Right. In ruby, module names are just constants. They can be assigned to variables and passed around. So you could create a new class, maybe something like `FHIRVersionIndependentRecord`. In the class, you can make your method `self.quality_of_life_observation(observation, fhir_record, patient, fhir_package)`. Then, in the code you can do things like `entry = fhir_package::Bundle::Entry.new`. Finally, in FhirRecord, you could have:

```ruby
def self.quality_of_life_observation(observation, fhir_record, patient)
  FHIRVersionIndependentRecord.quality_of_life_observation(observation, fhir_record, patient, FHIR)
end
```

You could repeat this for all of the duplicated code in FhirRecord and FhirDstu2Record. Although all of those changes would be for another pull request.